worksheet.rb

```diff
@@ -23,12 +33,138 @@
 # into this data structure, such as "DR0004"
 # and "3rd Feb 2016" and "RD0022"
 
+drivers = {
+  DR0004: [
+    {
+      cost: 5,
+      rider_id: 'RD0022',
+      rating: 5,
+      date: 'feb 3 2016'
+    },
+    {
+      cost: 10,
+      rider_id: 'RD0022',
+      rating: 4,
+      date: 'feb 4 2016'
+    },
+    {
+      cost: 20,
+      rider_id: 'RD0073',
+      rating: 5,
+      date: 'Feb 5 2016'
+    }
+  ],
+
+  DR0001: [
+    {
+      cost: 10,
+      rider_id: 'RD0003',
+      rating: 3,
+      date: 'Feb 3 2016'
+    },
+    {
+      cost: 30,
+      rider_id: 'RD0015',
+      rating: 4,
+      date: 'Feb 3 2016'
+    },
+    {
+      cost: 45,
+      rider_id: 'RD0003',
+      rating: 2,
+      date: 'Feb 5 2016'
+    }
+  ],
+  DR0002: [
+    {
+      cost: 25,
+      rider_id: 'RD0073',
+      rating: 5,
+      date: 'Feb 3 2016'
+    },
+    {
+      cost: 15,
+      rider_id: 'RD0013',
+      rating: 1,
+      date: 'Feb 4 2016'
+    },
+    {
+      cost: 35,
+      rider_id: 'RD0066',
+      rating: 3,
+      date: 'Feb 5 2016'
+    }
+  ],
+  DR0003: [
+    {
+      cost: 5,
+      rider_id: 'RD0066',
+      rating: 5,
+      date: 'Feb 4 2016'
+    },
+    {
+      cost: 50,
+      rider_id: 'RD0003',
+      rating: 2,
+      date: 'Feb 5 2016'
+    }
+  ]
+}
+
 ########################################################
 # Step 4: Total Driver's Earnings and Number of Rides
 
 # Use an iteration blocks to print the following answers:
-# - the number of rides each driver has given
-# - the total amount of money each driver has made
-# - the average rating for each driver
-# - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+
+puts 'The number of rides each driver has given:'
+drivers.each do |driver, rides|
+  puts "#{driver} - #{rides.length}"
+end
+
+puts 'The total amount of money each driver has made:'
+drivers.each do |driver, rides|
+  totaldriver = 0
+  rides.each do |info|
+    totaldriver += info[:cost]
+  end
+  puts "#{driver} - $#{totaldriver}"
+end
+
+puts 'The average rating for each driver:'
+drivers.each do |driver, rides|
+  totaldriver = 0
+  average = 0
+  rides.each do |info|
+    totaldriver = totaldriver.to_f + info[:rating]
+  end
+  average = totaldriver / rides.length
+  puts "#{driver} - #{average}"
+end
+
+puts 'Which driver made the most money?'
+drivercost = {}
+drivers.each do |driver, rides|
+  totaldriver = 0
+  rides.each do |info|
+    totaldriver += info[:cost]
+    drivercost[driver] = totaldriver
+  end
+end
```

Comment:
So, you have quite a few loops that break into these `.each` loops, with slightly different data in the body of each. That's a great time to break the contents of the bigger loops into methods!